server/api/users/users.test.ts

```diff
@@ -36,21 +37,108 @@ describe('auth router', () => {
     });
   });
 
-  describe('GET /', () => {
-    it('expects empty array if no users exist', async () => {
+  describe('GET /api/users', () => {
+    it('should have an API endpoint that exists', async () => {
       const res = await server.exec.get('/api/users/');
-      expect(res.body).toStrictEqual([]);
+
+      expect(res.status).toBe(200);
+    });
+
+    it('should return empty array if no users exist', async () => {
+      const res = await server.exec.get('/api/users/');
+
+      expect(res.body.users).toStrictEqual([]);
+    });
+
+    it('should return invalid page number with negative pages', async () => {
+      const res = await server.exec.get('/api/users/?page=-1');
+
+      expect(res.body.message).toEqual('Invalid page number');
+      expect(res.status).toBe(400);
+    });
+
+    it('should return invalid page numer with non-numerical pages', async () => {
+      const res = await server.exec.get('/api/users/?page=a');
+
+      expect(res.body.message).toEqual('Invalid page number');
+      expect(res.status).toBe(400);
     });
 
-    it('returns all users', async () => {
-      const userOne = await createUser();
-      const userTwo = await createUser();
+    it('should return page out of range with page that exceeds total pages', async () => {
+      const res = await server.exec.get('/api/users/?page=9001');
+
+      expect(res.body.message).toEqual('Page out of range');
+      expect(res.status).toBe(400);
+    });
 
+    it('should return proper page metadata with the first page and no users', async () => {
       const res = await server.exec.get('/api/users/');
-      expect(res.status).toBe(200);
-      expect(res.body[0]).toEqual(getExpectedUserObject(userOne));
-      expect(res.body[1]).toEqual(getExpectedUserObject(userTwo));
+
+      expect(res.body.page).toEqual(1);
+      expect(res.body.totalPages).toEqual(1);
+    });
+
+    it('should return proper page metadata with the first page and 20 users', async () => {
+      const count = 20;
+      await Promise.all(Array.from({ length: count }, createUser));
+
+      const res = await server.exec.get('/api/users/');
+
+      expect(res.body.page).toEqual(1);
+      expect(res.body.totalPages).toEqual(1);
     });
+
+    it('should return proper page metadata with the first page and 21 users', async () => {
+      const count = 21;
+      await Promise.all(Array.from({ length: count }, createUser));
+
+      const res = await server.exec.get('/api/users/');
+
+      expect(res.body.page).toEqual(1);
+      expect(res.body.totalPages).toEqual(2);
+    });
+
+    it('should return proper page metadata with the second page', async () => {
+      const count = 25;
+      await Promise.all(Array.from({ length: count }, createUser));
```

Comment:
if `count` isn't being reused anywhere we can just use count inline:

```suggestion
      await Promise.all(Array.from({ length: 25 }, createUser));
```